tests/test_contracts.py

```diff
@@ -503,3 +503,13 @@ def unary(arg: list[str]) -> None:
 
     with pytest.raises(AssertionError):
         unary(dict)
+
+
+def test_check_all_contracts_module_names_argument() -> None:
+    """Test that checks for classes and functions not declared as parameters to check_all_contracts
+    are skipped.
+    """
+    from tests.test_check_all_contracts.test_ignore_modules_not_in_arg import run
```

Comment:
One thing I forgot to mention is that pytest will recursively look for folders and files that contain `test_` (or something like that), so we don't want to include `test_` in names of folders or files that contain input data for other tests.

Naming is a bit tricky here. You could do something like `tests/fixtures/contracts/modules_not_in_arg.py`. ("Fixtures" is a term for objects used to set up a test context, which can include external files.)